config/config.go

```diff
@@ -1044,6 +1044,9 @@ type ConsensusConfig struct {
 	TimeoutPrecommit time.Duration `mapstructure:"timeout_precommit"`
 	// How much the timeout_precommit increases with each round
 	TimeoutPrecommitDelta time.Duration `mapstructure:"timeout_precommit_delta"`
+	// DelayedPrecommitTimeout how much time to wait before entering precommit starting
+	// from the commit time
+	DelayedPrecommitTimeout time.Duration // TODO dynamically change using the app
```

Comment:
Do you mind adding more of a description as to why it was introduced (i.e. to ensure a minimum block time)

Reply:
```suggestion
	// DelayedPrecommitTimeout how much time to wait before entering precommit (and sending precommit votes) starting
	// from the round time
	DelayedPrecommitTimeout time.Duration // TODO dynamically change using the app
```

Reply:
```suggestion
	// DelayedPrecommitTimeout ensures a minimum block time by waiting during the pre-commit time.
	// The new pre-commit vote time starts at: StartTime (the new block start time) + the delayed pre-commit time.
	//  Note that this change doesn't affect the rounds >= 1.
	DelayedPrecommitTimeout time.Duration // TODO dynamically change using the app
```